spec/initializer_spec.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../lib/initializer.rb'
+require_relative '../lib/board/board.rb'
+require_relative '../lib/board/board_checker.rb'
+require_relative '../lib/players/ai.rb'
+require_relative '../lib/players/human.rb'
+require_relative '../lib/players/player.rb'
+require_relative '../lib/ui/user_interface.rb'
+
+describe Initializer do
+  before(:each) do
+    board = Board.new(3, '-')
+    checker = BoardChecker.new(3, '-')
+    @user_interface = UserInterface.new(board, 'X', 'O')
+    @ai_logic = AI.new(checker, 'O', 'X')
+    @human_logic = Human.new(board, @user_interface, 'X')
+    @init = Initializer.new(3, '-')
+  end
+
+  describe 'setup_players' do
+    it 'sets up a game with human player as player 1 and computer as player 2' do
+      @init.setup_players('1', @ai_logic, @human_logic, @user_interface)
+      expect(@init.instance_variable_get(:@player1).instance_variable_get(:@behavior)).to be_a(Human)
+      expect(@init.instance_variable_get(:@player2).instance_variable_get(:@behavior)).to be_a(AI)
+    end
+
+    it 'sets up a game with computer player as player 1 and human as player 2' do
+      @init.setup_players('2', @ai_logic, @human_logic, @user_interface)
+      expect(@init.instance_variable_get(:@player1).instance_variable_get(:@behavior)).to be_a(AI)
+      expect(@init.instance_variable_get(:@player2).instance_variable_get(:@behavior)).to be_a(Human)
```

Comment:
This feels a little off to me. How can we test this without reaching into the class to inspect its "private" instance variables?